From 7949d67af674b86e5918ae412cb9be723a56c7da Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Wed, 22 Apr 2020 16:08:44 +0300
Subject: [PATCH] [SW-11702] fit: gaudi: set voltage according to module id

Set the infineon voltage according to different
module id, in case of mezzanines cards higher voltage is
required in order to support higher pll values

Change-Id: I3da3afd15f79fe7178a367b0322588d408067b2b
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/board/habanalabs/common/rootfs_overlay/etc/init.d/S85misc b/board/habanalabs/common/rootfs_overlay/etc/init.d/S85misc
index 4dbfa5e..c5f3508 100755
--- a/board/habanalabs/common/rootfs_overlay/etc/init.d/S85misc
+++ b/board/habanalabs/common/rootfs_overlay/etc/init.d/S85misc
@@ -24,13 +24,31 @@
 	fi
 }
 
+# voltage differs between devices due to different pll setups
+function voltage_set()
+{
+	module_id=$1
+
+	if [ -z $module_id ]; then
+		echo "voltage_set(): module id wasn't supplied"
+		return
+	fi
+
+	voltage_path=`find /sys/ -name "in3_cmd"`
+	if [ $module_id -eq 0 ]; then
+		echo 840 > $voltage_path
+	else
+		echo 860 > $voltage_path
+	fi
+}
+
 function module_id_set()
 {
 	device=`hostname`
 	case "$device" in
 	gaudi)
 		[ ! -e $module_id_file ] && return
-		module_id=$(printf "%d" 0x`xxd -p $module_id_file`)
+		local module_id=$(printf "%d" 0x`xxd -p $module_id_file`)
 		if [ $module_id -gt 0 ]; then
 			mezz_id=$((module_id - 1))
 			set_message_of_the_day $mezz_id
@@ -42,6 +60,8 @@
 	*)
 	;;
 	esac
+
+	echo "$module_id"
 }
 
 function start()
@@ -49,7 +69,8 @@
 	echo "Starting misc..."
 	echo 1 > /sys/module/rcupdate/parameters/rcu_cpu_stall_suppress
 	/usr/sbin/eeprom_dump
-	module_id_set
+	module_id=$(module_id_set)
+	voltage_set $module_id
 }
 
 function stop()
